### PR TITLE
Fix switch `default` execution

### PR DESCRIPTION
**Pull request by HalidOdat**
Fixes switch `default` case execution, when a default case has regular switch cases below it (as in this test262 [test](https://github.com/tc39/test262/blob/72c0c5e16350a76bd41f7a1ceb7702588a2a39c6/test/language/statements/switch/S12.11_A1_T2.js) ) we execute the default case body at the end, we should execute the cases in order.

### CODE REVIEW

File: boa_engine/src/bytecompiler/mod.rs [hunk 1]
```diff
@@ -295,6 +295,7 @@ pub struct ByteCompiler<'ctx, 'host> {
 impl<'ctx, 'host> ByteCompiler<'ctx, 'host> {
     /// Represents a placeholder address that will be patched later.
     const DUMMY_ADDRESS: u32 = u32::MAX;
+    const DUMMY_LABLE: Label = Label { index: u32::MAX };
```
Comment (raskad) [inline boa_engine/src/bytecompiler/mod.rs:298]:
```suggestion
    const DUMMY_LABEL: Label = Label { index: u32::MAX };
```

**Review by raskad — APPROVED** [review]
Nice work! Just got a typo :)

File: boa_ast/src/statement/switch.rs [hunk 2]
```diff
@@ -31,14 +31,30 @@ impl Case {
...
     #[must_use]
     pub const fn new(condition: Expression, body: StatementList) -> Self {
-        Self { condition, body }
+        Self {
+            condition: Some(condition),
+            body,
+        }
+    }
+
+    /// Creates a `Case` AST node.
```
Comment (nekevss) [inline boa_ast/src/statement/switch.rs:40]: Can this doc highlight that this is a default case?

**Review by nekevss — COMMENTED** [review]
Looks pretty good to me 😄 Just one doc nitpick.

**Review by nekevss — APPROVED** [review]
LGTM 😄 Nice work!

**Pull request merged**